### DISCUSSION TITLE
[4.0] Change behaviour of toolbar in TinyMCE

### DISCUSSION THREAD

**Discussion by drmenzelit** [joomla/joomla-cms#32090] [Q&A] [answered]
In TinyMCE the second line of icons in the toolbar is hidden until the user click on the 3 dots:
![grafik](https://user-images.githubusercontent.com/9153168/105207548-ef44c800-5b47-11eb-83e0-b04d9d6e7d5c.png)
![grafik](https://user-images.githubusercontent.com/9153168/105207588-fbc92080-5b47-11eb-9b53-dc116868dde5.png)

In glip a user asked if it possible to change this behaviour to have the toolbar always open or have a parameter to change it (like it is in JCE for example).
I found the line in plugins/editors/tinymce/tinymce.php that controls the sliding, line 614:
'toolbar_mode'       => 'sliding',
changing it to 'wrap' makes the toolbar be open. But I don't know if it is possible to have a parameter for that.

**Comment by brianteeman** [comment, marked as answer]
Yes its possible. There are 4 possible settings that can easily be added as a paramater
https://www.tiny.cloud/docs/configure/editor-appearance/#toolbar_mode

**Comment by richard67** [comment]
@drmenzelit Does that answer above solve your question? If so, you can mark Brian's answer above as answer, so the discussion is shown as solved.

**Reply by drmenzelit** [reply]
Yes, that answers the question. Now I have to look if I can create a PR for that :-)

**Reply by brianteeman** [reply]
I can do it if you want

**Reply by drmenzelit** [reply]
That would be great, thank you Brian

**Reply by brianteeman** [reply]
Here you are #32091
